juju/juju
# Use stateful set for pods with storage

## Description of change

For units requiring storage, use a k8s StatefulSet instead of a Deployment Controller to manage the pods.
This PR required a regeneration of the mocks using go generate so ignore those file when reviewing.

The basis is to use a deployment controller for pods without storage and a statefulset for those with. The storage setup logic is essentially the same, but is tweaked a bit because the pv claim creation is not done  manually for stateful sets.

## QA steps

Deploy a test k8s charm with storage. Ensure a pv and pvc is properly set up and the specified mount point used.
